public/apps/configuration/panels/internal-user-edit/internal-user-edit.tsx

```diff
@@ -142,19 +145,23 @@ export function InternalUserEdit(props: InternalUserEditDeps) {
         </EuiFlexGroup>
       </EuiPageHeader>
       <PanelWithHeader headerText="Credentials">
-        <EuiForm>
+        <EuiForm isInvalid={showErrors} error={errors}>
           <FormRow
             headerText="Username"
             headerSubText="Specify a descriptive and unique user name. You cannot edit the name once the user is created."
-            helpText="The username must contain from m to n characters. Valid characters are
-             lowercase a-z, 0-9 and (-) hyphen."
+            helpText={resourceNameHelpText('user')}
+            isInvalid={showErrors}
           >
             <EuiFieldText
               value={userName}
               onChange={(e) => {
                 setUserName(e.target.value);
               }}
+              onBlur={() => {
+                checkForResourceNameErrors('user', userName);
+              }}
               disabled={props.action === 'edit'}
+              isInvalid={showErrors}
```

Comment:
Shall we consider creating a shared component for it?

Reply:
Yes. Created shared component named `NameRow`